packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,17 +6,87 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, isSFDXContainerMode } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+type CLIOrgData = {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+};
+
+class ForceOrgOpenContainerExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
+      .withArg('force:org:open')
+      .withLogName('force_org_open_default_scratch_org')
+      .withArg('--urlonly')
+      .withJson()
+      .build();
+  }
+
+  private buildUserMessageWith(orgData: CLIOrgData): string {
+    return nls.localize(
+      'force_org_open_container_mode_message_text',
+      orgData.result.orgId,
+      orgData.result.username,
+      orgData.result.url
+    );
+  }
+
+  private openBrowserWithCliResponse(cliResponseJSON: string | Buffer) {
+    try {
+      const cliOrgData: CLIOrgData = JSON.parse(cliResponseJSON.toString());
+      const authenticatedOrgUrl: string = cliOrgData.result.url;
+
+      channelService.appendLine(this.buildUserMessageWith(cliOrgData));
+
+      if (authenticatedOrgUrl) {
+        vscode.env.openExternal(vscode.Uri.parse(authenticatedOrgUrl));
+      }
+    } catch (e) {
+      telemetryService.sendErrorEvent(e.message, e.stack);
+    }
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
```

Comment:
I think you need to add `env: { SFDX_JSON_TO_STDOUT: 'true' }` here so that you also get the error results from the cli in `execution.stdoutSubject.subscribe`

Reply:
Another good catch. This is again another place where the contract is broken from https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts#L49

Reply:
Sorry, I was looking at `baseDeployCommand` which uses that. But the normal command in https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/commands/commands.ts#L326 doesn’t use that.

@lcampos - Do all commands need to use that? In that case, it needs to be changed in the base command?